### ISSUE TITLE
[1.5.0, static libs] Mixed Objective-C/Swift pod fails to import Swift from Objective-C

### ISSUE THREAD

**Issue by MaximeLM** [CocoaPods/CocoaPods#7594]
* [🌈] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

First, a lot of thanks for this feature ❤️

# Report

## What did you do?

* Update to CocoaPods 1.5.0
* Remove `use_frameworks!` from Podfile

Pod setup: one Pod `TestLib` with mixed Swift/Objective-C code. Objective-C files import Swift code with `#import <TestLib/TestLib-Swift.h>`.

## What did you expect to happen?

The project would build.

## What happened instead?

The project failed to build with error `'TestLib/TestLib-Swift.h' file not found`.

### Stack

```
   CocoaPods : 1.5.0
        Ruby : ruby 2.2.1p85 (2015-02-26 revision 49769) [x86_64-darwin14]
    RubyGems : 2.4.6
        Host : Mac OS X 10.13.4 (17E199)
       Xcode : 9.3 (9E145)
         Git : git version 2.15.1 (Apple Git-101)
Ruby lib dir : /Users/maximelemoine/.rvm/rubies/ruby-2.2.1/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 4ddb0bcf3b52ed9ff3e40a475fb06d9dc264bcfe
```

### Installation Source

```
Executable Path: /Users/maximelemoine/.rvm/gems/ruby-2.2.1/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-packager    : 1.5.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
platform :osx, '10.10'

#use_frameworks!

target 'TestSwiftImport' do
  
  pod 'TestLib', :path => '.'

end
```

## Project that demonstrates the issue

https://github.com/MaximeLM/TestSwiftImport

**Comment by dnkoutso** [comment]
This is a known issue you can workaround it via `#import "TestLib-Swift.h"` instead.

**Comment by Zmeu213** [comment]
... or just use cocoapods 1.4

**Comment by dnkoutso** [comment]
sure, you do you @Zmeu213

**Comment by Zmeu213** [comment]
@dnkoutso (:

**Comment by MaximeLM** [comment]
@dnkoutso ok, I'll use the workaround

**Comment by nolanw** [comment]
@dnkoutso you say this is a "known issue", is there an issue open already?

**Comment by dnkoutso** [comment]
@MaximeLM apologies meant meant internally and we should keep this open.

**Comment by mdsb100** [comment]
Same Issue +1

@MaximeLM 
@dnkoutso 

Did you realize that the more serious problem is:  As a pod author, you can not limit user how to use!
```
use_frameworks!
//to be
#import <TestLib/TestLib-Swift.h>
```

```
//remove use_frameworks!
//to be
#import "TestLib-Swift.h"
```

So the solution is:
```
#ifdef  AMacro
#import "TestLib-Swift.h"
#else  
#import <TestLib/TestLib-Swift.h>
#endif

//And Find ways to inject 'AMacro' into your application
```

Another serious problem is:
When using swift, the lint function Cocoapods provide is compiled to dynamic framework?Why not lint by static framework? 


Although using macros can solve this problem. When lint is a dynamic library, use `<>`. In the other case, `""` is used.

**Comment by mdsb100** [comment]
@orta Thank you，I learned.

I think this issue could be close.

**Comment by nolanw** [comment]
I disagree, this issue should remain open. The `__has_include` workaround is helpful, but forcing pod authors to deviate from Apple docs doesn’t seem great if it’s possible for CocoaPods to configure build settings appropriately. The ideal fix is to have libraries build as-is.

If that’s not possible then some documentation is probably warranted.

But I didn’t open the issue so I can’t exactly go making demands everywhere :)

**Comment by dnkoutso** [comment]
Will keep it open.

**Comment by mdsb100** [comment]
@nolanw 
I think it is so hard ... But I agree you said!
I hate protobuf by using `use_framework!`. I use swift so I have to `use_framwork` until the release of cocapods@1.5. 

I can finally avoid using the [disgusting macro (`GPB_USE_PROTOBUF_FRAMEWORK_IMPORTS `)](https://github.com/google/protobuf/blob/master/objectivec/google/protobuf/Duration.pbobjc.m) to use both swift and objc-protobuf. There is some issue so I can not use swift-protobuf.

So the protobuf author must be very happy to hear: cocoapods can make libraries build as-is!

**Comment by MaximeLM** [comment]
@orta thanks! What I end up using in our private pod is:
```
#if __has_include("TestLib-Swift.h")
    #import "TestLib-Swift.h"
#else
    #import <TestLib/TestLib-Swift.h>
#endif
```
Of course, this workaround is only possible for pod authors.

**Comment by mxcl** [comment]
Is there a final say on this issue yet?

I'm not a fan of adding the hack, since I've already in this life experienced situations where such hacks break other situations, and considering this world of mixed package managers, unexpected things happen.

But, if that's the only way this is going to work, I'll do it.

**Comment by segiddins** [comment]
We haven't found a better way yet

**Comment by AaronLBratcher** [comment]
I think I'm a victim of the same thing and want to confirm.

I have an internal framework I'm adding as a pod to my project. This framework is a mixture of Swift and Objective-C (same as the project.)

When adding the internal pod to my project, it shows as a Development Pod in my workspace. Any attempt to import the swift.h from my Objecetive-C files results in a file not found. I looked and found that there is no -Swift.h file in the created framework package.

**Comment by ChiragBhutani1988** [comment]
@dnkoutso I am trying to include socket.io swift client using cocoapods in my existing project which has already Objective C pods and my project is in Objective C. When I do so I get the error "module map file '/Users/.../Library/Developer/Xcode/DerivedData/BCFramework-aecawjeshbaibfeekdiasoxrrcjz/Build/Products/Debug-iphoneos/Socket.IO-Client-Swift/SocketIO.modulemap' not found"

**Comment by amorde** [comment]
@ChiragBhutani1988 what you're describing sounds like a different issue - could you open another issue with a sample project + steps to repro? thanks

**Comment by fer662** [comment]
@amorde It appears this is broken in cocoapods 1.5.3. I was having trouble getting it to work in the attached project with the proposed workaround, and so i tried to revert to 1.5.0 via gemfile. After a clean and rebuild it worked. Going back to 1.5.3, clean, build, doesn't work again.

**Comment by amorde** [comment]
@fer662 what exactly is broken? there are a few things discussed in this issue.

**Comment by fer662** [comment]
@amorde sorry for being unspecific. The import for for "LibName-Swift.h" works on 1.5.0, but results in a file not found in 1.5.3, where LibName is a swift pod integrated as a static library.

**Comment by davidahouse** [comment]
Noticing the same problem with 1.6.1. Create an empty ObjC project, then add a Swift pod and do not set use_frameworks!. Now try to #import "LibName-Swift.h" and you get not found. I tried with a couple of different Pods including AlamoFire. Interestingly enough I do see the "LibName-Swift.h" file in the build folder, but for some reason it isn't found by Xcode. So the solution I assume is to get this file generated/moved to the correct folder to be found, OR make a change to the project so it can find it.

**Comment by komocode** [comment]
Tried #import "PromiseKit-Swift.h", doesn't work. Any other workarounds?

**Comment by abbeycode** [comment]
I'm also seeing that the Library-Swift.h file still isn't being generated for static libs on CP v1.8.4. Without it, I'm unable to use the pod as a static lib. Any idea why the workaround from 1.5.0 isn't working anymore?

**Comment by aphex3k** [comment]
> @dnkoutso I am trying to include socket.io swift client using cocoapods in my existing project which has already Objective C pods and my project is in Objective C. When I do so I get the error "module map file '/Users/.../Library/Developer/Xcode/DerivedData/BCFramework-aecawjeshbaibfeekdiasoxrrcjz/Build/Products/Debug-iphoneos/Socket.IO-Client-Swift/SocketIO.modulemap' not found"
> 
> One thing to note here is that I am creating a custom framework instead of an application. Please suggest how should I proceed.

I have the same issue but can't find if @ChiragBhutani1988 has create a separate issue or not...

**Comment by vbabenkoru** [comment]
@komocode @abbeycode Has anyone been able to figure this out? All the workarounds are for pods with mixed Swift/Objective-C code. It seems like almost no one is trying to use Swift static libraries from an Objective-C project, which has no workarounds, apparently.

**Comment by MJ8393** [comment]
You can mix swift and objective-C in Framework, not pod library. Take a look at here: https://jsorge.net/2019/01/23/mixed-language-framework
